### ISSUE TITLE
`bors blocker` command

### ISSUE THREAD

**Issue by matklad** [bors-ng/bors-ng#157]
This is a nice-to-have feature request.

Even under the stern bors guidance, the master can still be broken if something external to the build itself changes (for example, the build uses the `nightly` compiler, and the next nightly breaks things). In such a case, several pull requests may fail to pass the checks. 

So you have to send a pull requests that fixes master, wait for it to be merged and then manually retry all the pending ones. It would be great to say something like `bors retry all` and have the bot do this automatically. Perhaps this can be extended even to something like `bors blocker` which means that until this PR is merged, no other should be in the queue, and that all `r+` should be buffered until the blocker is resolved.

**Comment by Macarse** [comment]
@matklad can you explain a bit more the example?

What confuses me is that you said that something external breaks master but it can be fixed sending  a PR into your own master? What would that change be? forcing a version of a dependency for example?

**Comment by matklad** [comment]
>What would that change be? forcing a version of a dependency for example?

Yep, that would be a possible example. Or something along the lines of "new nightly compiler thinks that code XYZ is illegal, so I replace it with ZXX instead".

Something like this actually happend in IntelliJ Rust (https://github.com/intellij-rust/intellij-rust/pull/1140). 

In tests, we check that our parser is able to parse sources of the rust compiler itself, and we do test nightly versions of the compiler because we need to support new syntax as soon as it arrives at nightly. So recently the attributes on fields of the struct expressions had been introduced, which was not supported in our parser and make a couple of pull requests to fail on r+. I've fixed that in 1140, and then r+ those pull requests again.

**Comment by mythmon** [comment]
In mozilla/normandy, for part of our CI we use an external build system (Firefox's), but prepare a build environment for it within our repo. Occasionally the requirements of the external build system change, and we have to update our environment to match that. These changes would fit a `bors blocker` or similar command well. We ran into this kind of issue in mozilla/normandy#1149 and fixed the blocker in mozilla/normandy#1150.
